controller/app_controller.go

```diff
@@ -0,0 +1,60 @@
+package controller
+
+import (
+    "fmt"
+    "log"
+    "net/http"
+    "strconv"
+    "encoding/json"
+
+
+    "pokedex/service"
+    "pokedex/model"
+
+    "github.com/gorilla/mux"
+)
+
+/* Used to initialize Pokedex Services */
+var pokedex = service.Pokedex(make(map[int]model.Pokemon))
+
+/* Obtains pokemon by id and returns it in JSON string format */
+func get_pokemon_by_id(w http.ResponseWriter, r *http.Request) {
```

Comment:
Go comments above functions, methods, structs, interfaces, variables, etc. normally are added with double backslash, even if they are multi-line. This is so because Go comments over these are actually used to document your project.

Take a look at the following example: https://golang.org/pkg/fmt/#Println
And the source code of that function: https://golang.org/src/fmt/print.go?s=7957:8006#L263

These comments are done in the following format (using function as an example):
`// <name_of func> <description>`

Having a space between the `//` and the `name_of_func`, and another space between `<name_of_func> and <description>.

Also, do not use `snake_case` to name identifiers, Go encourages the use of camelCase or PascalCase (depending on the scope of the identifier).

It should be changed to something like this:
```suggestion
// GetPokemonByID obtains pokemon by id and returns it in JSON string format
func GetPokemonByID(w http.ResponseWriter, r *http.Request) {
```